File: src/Aneejian.Games.ClickMatch/Data/LevelStatsDTO.cs
namespace Aneejian.Games.ClickMatch.Data;

public class LevelStatsDTO
{
	public int UserId { get; set; }
	public int Level { get; set; }
	public int HighScore { get; set; }
	public int TimesPlayed { get; set; }
	public int TimesWon { get; set; }
	public int TimesLost { get; set; }
	public int TimesAbandoned => TimesPlayed - (TimesWon + TimesLost);
	public GameDto? UsersBestGame { get; set; }
	public GameDto? LevelsBestGame { get; set; }
}

File: src/Aneejian.Games.ClickMatch/wwwroot/js/LevelStatsDTO.js

export class LevelStatsDTO {
    constructor(levelData) {
        this.userId = levelData.userId;
        this.level = levelData.level;
        this.highScore = levelData.highScore;
        this.timesPlayed = levelData.timesPlayed;
        this.timesWon = levelData.timesWon;
        this.timesLost = levelData.timesLost;
        this.usersBestGame = levelData.usersBestGame;
        this.levelsBestGame = levelData.levelsBestGame;
    }
}
